app/controllers/classified_listings_controller.rb

```diff
@@ -64,27 +70,22 @@ def create
   def update
     authorize @classified_listing
 
-    # NOTE: this should probably be split in three different actions: bump, unpublish, publish
-    if listing_params[:action] == "bump"
-      cost = ClassifiedListing.cost_by_category(@classified_listing.category)
-
-      org = Organization.find_by(id: @classified_listing.organization_id)
+    cost = ClassifiedListing.cost_by_category(@classified_listing.category)
 
-      available_org_credits = org.credits.unspent if org
-      available_user_credits = current_user.credits.unspent
+    # NOTE: this should probably be split in three different actions: bump, unpublish, publish
+    return bump_listing(cost) if listing_params[:action] == "bump"
 
-      if org && available_org_credits.size >= cost
-        charge_credits_before_bump(org, cost)
-      elsif available_user_credits.size >= cost
-        charge_credits_before_bump(current_user, cost)
-      else
-        redirect_to(credits_path, notice: "Not enough available credits") && return
-      end
-    elsif listing_params[:action] == "unpublish"
+    if listing_params[:action] == "unpublish"
       unpublish_listing
+      redirect_to "/listings/dashboard"
+      return
     elsif listing_params[:action] == "publish"
+      if @classified_listing.bumped_at.nil?
+        return first_publish(cost)
```

Comment:
In my opinion `return first_publish(cost)` it's a bit misleading, we are not actually returning the result of the `first_publish`, but executing `first_publish` and returning.
You can also check `bumped_at` like this here: `@classified_listing.bumped_at?`